### DISCUSSION TITLE
How do you extract a JSON property value from an HTTP source with unstructured JSON response?

### DISCUSSION THREAD

**Discussion by trevorstr** [ArroyoSystems/arroyo#656] [General]
- I launched Arroyo as a container
- I used the web interface to create an HTTP connect **source** with unstructured JSON schema response
- I tried to access the `value` field, but I just get an error saying it's an invalid field name

```
SELECT extract_json_string(value, '$.species.name') FROM http01;
```

<img width="1280" alt="Screenshot 2024-06-10 at 6 45 50 PM" src="https://github.com/ArroyoSystems/arroyo/assets/147647647/703143e8-621d-412e-979f-d6db87d54929">

When I created the HTTP connection source, it said that there would be a `value` field that I could access the JSON from. So how do I access the JSON response from the HTTP source, in my `SELECT` query?

**Comment by mwylde** [comment]
Hey @trevorstr — thanks for trying Arroyo!

If you expand the http01 source in the sources tab (clicking the v icon), it will show you the schema. If you created it with unstructured JSON, it should have a single `value JSON` field. If you see something else, I would try recreating the source and check that these options are selected:

![image](https://github.com/ArroyoSystems/arroyo/assets/100920/c90b2140-cbb1-484e-a72f-276a36fa3cd4)

**Comment by trevorstr** [comment]
@mwylde Thanks. Those are the exact options I selected, but there are no schema field entries. I created a duplicate HTTP source, but the same problem persisted.

**Reply by mwylde** [reply]
Hm, that's interesting. I'm not able to reproduce. What version of arroyo is this? (I.e., if you're running in docker, what's the tag of the image you're running?)

In the meantime you can work around this by creating the source in SQL instead of using the webui: (see the example at the bottom of https://doc.arroyo.dev/connectors/polling-http#configuring-the-connection):

```sql
create table http01 (
  value JSON
) with (
  connector = 'polling_http',
  endpoint = 'http://example.com',
  format = 'json',
  'json.unstructured' = 'true'
);
```

**Reply by trevorstr** [reply]
The container image hash part is `3fd812f4c149`.

I just created another HTTP data source, and this time it has the `value` field. I have no idea why it was failing last night.
